frontend/src/components/ui/chat/ChatList.js

```diff
@@ -1,19 +1,47 @@
-import Chat from "./Chat";
-import styles from "./ChatList.module.css";
+import { useRef, useEffect } from 'react';
+
+import { v4 as uuidv4 } from 'uuid';
+
+import Chat from './Chat';
+import styles from './ChatList.module.css';
+
+const AlwaysScrollToBottom = () => {
+  const messageEndRef = useRef();
+  /*
+    false argument passed to scrollIntoView so that
+    the bottom of the element will be aligned to 
+    the bottom of the visible area of the scrollable ancestor
+  */
+  useEffect(() => messageEndRef.current.scrollIntoView(false));
+  return <div ref={messageEndRef} />;
+};
+
+function ChatList({ chats }) {
+  let lastChatMessage = chats[chats.length - 1];
 
-function ChatList(props) {
   return (
     <ul className={styles.chat_list}>
-      {/*For each message objects, create the HTML view of the msg id, sender id and the message */}
-      {props.chats.map((chat) => {
+      {/* For each chat object, create the HTML view of the msg id, sender id and the message */}
+      {chats.map((chat) => {
+        let uniqueChatKey = uuidv4();
         return (
-          <li key={chat.senderId}>
-            <Chat text={chat.text} senderId={chat.senderId} />
+          <li key={uniqueChatKey}>
```

Comment:
Personally, I think `uuidv4()` is simple enough to not be abstracted out as a constant